tsdb/store.go

```diff
@@ -1496,7 +1522,19 @@ func (s *Store) DeleteSeries(database string, sources []influxql.Source, conditi
 		// No series file means nothing has been written to this DB and thus nothing to delete.
 		return nil
 	}
-	shards := s.filterShards(byDatabase(database))
+
+	shardFilterFn := byDatabase(database)
+	if len(sources) != 0 {
+		if measurement, ok := sources[0].(*influxql.Measurement); ok {
+			if measurement.RetentionPolicy != "" {
+				shardFilterFn = ComposeShardFilter(shardFilterFn, byRetentionPolicy(measurement.RetentionPolicy))
+			}
+		} else {
+			return fmt.Errorf("unsupported source type in delete %v", sources[0])
+		}
+	}
```

Comment:
I think we decided the parser wouldn't allow more than a single source, but I think it would still be good to check for that here and return an error. There might be another way to get more than measurement (Flux? Probably not, but are you sure?). There also might be a change made to the parser that allows more than one source, and if we don't check for that possibility here that could lead to strange issues in the future that we spend weeks tracking down. Returning an error instead what make it clear what happened.